### DISCUSSION TITLE
Bigger webp in size when converting from jpeg

### DISCUSSION THREAD

**Discussion by hien** [libvips/libvips#3147] [Help wanted]
Hi,

I tried convert this jpeg image to webp. But output webp file size is bigger:

```
root@dev-box-01:~# vips copy Coffee.jpg Coffee.webp
root@dev-box-01:~# ls -lah Coffee.jpg
-rw-r--r-- 1 root root 125K Nov  5 15:00 Coffee.jpg
root@dev-box-01:~# ls -lah Coffee.webp
-rw-r--r-- 1 root root 139K Nov  7 17:32 Coffee.webp
root@dev-box-01:~# file Coffee.jpg
Coffee.jpg: JPEG image data, Exif standard: [TIFF image data, little-endian, direntries=0], baseline, precision 8, 1920x1440, components 3
root@dev-box-01:~# file Coffee.webp
Coffee.webp: RIFF (little-endian) data, Web/P image
root@dev-box-01:~# vips -v
vips-8.12.1-Wed Feb  2 14:43:28 UTC 2022
```

Image in this discussion can be downloaded at https://drive.google.com/file/d/19tTEPvnYbiI6Z2N8wtmh1rk6T4JJVoiU/view?usp=share_link

Anyone has similar this issues?

Thank you,

Hien

**Comment by jcupitt** [comment]
Hello @hien,

Try this:

```
$ vips copy Coffee.jpeg x.webp
$ vips copy Coffee.jpeg x.jpg
$ ls -l x*
-rw-r--r-- 1 john john 191564 Nov  7 11:35 x.jpg
-rw-r--r-- 1 john john 141444 Nov  7 11:35 x.webp
$ ls -l Coffee.jpeg 
-rw-rw-r-- 1 john john 127604 Nov  7 11:32 Coffee.jpeg
```

So the webp is smaller than jpeg compressed by libvips, it's just not as small as the original `Coffee.jpg`.

I think your test image has been compressed to a very low quality level, or perhaps has been compressed with mozjpeg.

You can set Q for webp, eg.:

```
$ vips copy Coffee.jpeg x.webp[Q=50]
$ ls -l x.webp
-rw-r--r-- 1 john john 108538 Nov  7 11:38 x.webp
```

Now it's smaller again, though of course quality may be worse.

AVIF gives:


```
$ vips copy Coffee.jpeg x.avif
$ ls -l x.avif
-rw-r--r-- 1 john john 72596 Nov  7 11:39 x.avif
```

Smaller still.

**Reply by hien** [reply]
Hello @jcupitt

Thanks for your reply.

You're right. I checked this image via EXIF data, it showed the quality level is at 10%.

So the only workaround for webp smaller size (skipping quality)  now is using Q?

Hien.

**Reply by jcupitt** [reply]
I think it depends on the image type. webpsave has quite a few controls:

https://www.libvips.org/API/current/VipsForeignSave.html#vips-webpsave
